### PR TITLE
Use depset for SDK files

### PR DESCRIPTION
**Pull request by dzbarsky**
This PR improves analysis phase time by using depsets.

Example private repo, note `go_test_impl` and `go_binary_impl` disappear from the profile and `go_library_impl` contracts.

Before
image

After
image

### CODE REVIEW

**Comment by fmeum** [comment]
Could you also update the docs (`providers.rst`, I think)?

**Comment by dzbarsky** [comment]
> Could you also update the docs (`providers.rst`, I think)?

yep, done

File: go/private/actions/stdlib.bzl [hunk 1]
```diff
@@ -56,7 +56,7 @@ def _should_use_sdk_stdlib(go):
     if version and version[0] <= 1 and version[1] <= 19 and go.sdk.experiments:
         # The precompiled stdlib shipped with 1.19 or below doesn't have experiments
         return False
-    return (go.sdk.libs and  # go.sdk.libs is non-empty if sdk ships with precompiled .a files
+    return (go.sdk.libs.to_list() and  # go.sdk.libs is non-empty if sdk ships with precompiled .a files
```
Comment (fmeum) [inline go/private/actions/stdlib.bzl:59]:
```suggestion
    return (go.sdk.libs and  # go.sdk.libs is non-empty if sdk ships with precompiled .a files
```
should work as well

**Review by fmeum — APPROVED** [review]

File: go/private/actions/stdlib.bzl (hunk 1 above)
Comment (dzbarsky) [reply go/private/actions/stdlib.bzl:59]: ah neat I wasn't sure if an empty depset would be falsy

**Review by fmeum — APPROVED** [review]

**Pull request merged**